src/test/java/org/reaktivity/nukleus/tcp/internal/streams/ClientPartialWriteLimitsIT.java

```diff
@@ -92,9 +92,10 @@
     })
     public void shouldWriteWhenMoreDataArrivesWhileAwaitingSocketWritableWithoutOverflowingSlot() throws Exception
     {
-        ProcessDataHelper.fragmentWrites(IntStream.of(5, 6));
-        AtomicBoolean allDataWritten = new AtomicBoolean(false);
-        HandleWriteHelper.fragmentWrites(generate(() -> allDataWritten.get() ? ALL : 0));
+        AtomicInteger dataFramesReceived = new AtomicInteger();
+        ProcessDataHelper.fragmentWrites(generate(() -> dataFramesReceived.incrementAndGet() == 1 ? 5
+                : dataFramesReceived.get() == 2 ? 6 : ALL));
+        HandleWriteHelper.fragmentWrites(generate(() -> dataFramesReceived.get() >= 2 ? ALL : 0));
```

Comment:
Is it fair to say that `dataFramesReceived` is effectively tracking the change from `6` to `ALL` in the `[5, 6, ALL*]` sequence of `socket.write(...)` return values in `processData(...)` and is then being used to trigger the change from `0` to `ALL` in the `[0*, ALL*]` sequence of `socket.write(...)` return values in `handleWrite(...)`?